components/TARGET_PSA/services/crypto/COMPONENT_PSA_SRV_IPC/psa_crypto_spm.c

```diff
@@ -1077,55 +888,56 @@ psa_status_t psa_generate_random(uint8_t *output,
 psa_status_t mbedtls_psa_inject_entropy(const unsigned char *seed,
                                         size_t seed_size)
 {
-    psa_status_t status;
-    psa_handle_t handle = PSA_NULL_HANDLE;
-    psa_invec in_vec = { seed, seed_size };
-
-    handle = psa_connect(PSA_ENTROPY_ID, MINOR_VER);
-    if (handle <= 0) {
-        return (PSA_ERROR_COMMUNICATION_FAILURE);
-    }
-
-    status = psa_call(handle, &in_vec, 1, NULL, 0);
-    psa_close(handle);
+    psa_invec in_vec[1] = {
+        { seed, seed_size }
+    };
 
+    psa_status_t status = ipc_oneshot(PSA_ENTROPY_ID, in_vec, 1, NULL, 0);
     return (status);
 }
+
 /****************************************************************/
 /* PSA Generator */
 /****************************************************************/
+
 psa_status_t psa_get_generator_capacity(const psa_crypto_generator_t *generator,
                                         size_t *capacity)
 {
-    psa_status_t status;
     psa_crypto_derivation_ipc_t psa_crypto_ipc = { 0, 0, 0, 0 };
     psa_crypto_ipc.func = PSA_GET_GENERATOR_CAPACITY;
-    psa_invec in_vec = { &psa_crypto_ipc, sizeof(psa_crypto_ipc) };
-    psa_outvec out_vec = { capacity, sizeof(*capacity) };
 
-    if (generator->handle <= 0) {
-        return (PSA_ERROR_BAD_STATE);
-    }
+    psa_invec in_vec[1] = {
+        { &psa_crypto_ipc, sizeof(psa_crypto_ipc) }
+    };
 
-    status = psa_call(generator->handle, &in_vec, 1, &out_vec, 1);
+    psa_outvec out_vec[1] = {
+        { capacity, sizeof(*capacity) }
+    };
+
+    psa_status_t status = ipc_call((psa_handle_t *)&generator->handle, in_vec, 1, out_vec, 1, false);
```

Comment:
don't need the cast

Reply:
const to non-const, this produced a warning on GCC